File: src/main/java/mil/nga/tiff/field/type/enumeration/SampleFormat.java
package mil.nga.tiff.util;

import java.util.Arrays;

public enum SampleFormat {
    UNSIGNED_INT(1),
    SIGNED_INT(2),
    FLOAT(3),
    UNDEFINED(4);

    private final int id;

    SampleFormat(int id) {
        this.id = id;
    }

    public static SampleFormat findById(int id) {
        return Arrays.stream(values())
            .filter(o -> o.id == id)
            .findAny()
            .orElseThrow(() -> new TiffException("Invalid ID: " + id));
    }

    public int getId() {
        return id;
    }
}

File: src/main/java/mil/nga/tiff/field/type/enumeration/SampleFormat.kt
package mil.nga.tiff.field.type.enumeration

enum class SampleFormat(val id: Int) {
    UNSIGNED_INT(1),
    SIGNED_INT(2),
    FLOAT(3),
    UNDEFINED(4);

    companion object {

        @JvmField
        public val DEFAULT = UNDEFINED

        @JvmStatic
        fun findById(id: Int?): SampleFormat {
            if (id == null) {
                return DEFAULT
            }

            return entries.first { it.id == id}
        }
    }
}
